### PR TITLE
prepare-config: Remove flags from CLI

### PR DESCRIPTION
**Pull request by oliverwm1**
Remove flag options from `prepare-config` CLI. Specifically `--model_url` and `--diagnostics_ml`.

Refactored public API:
- the `prepare-config` CLI now takes only a path to a YAML, instead of the YAML and optional flags.

Significant internal changes:
- modified the e2e argo workflow to patch the prognostic config as required

- [x] Tests added

### CODE REVIEW

File: workflows/prognostic_c48_run/runtime/segmented_run/prepare_config.py [hunk 1]
```diff
@@ -197,26 +196,6 @@ def _create_arg_parser() -> argparse.ArgumentParser:
...
-            "Remote url to a trained ML model. If a model is omitted (and not "
-            "specified in `user_config`'s `scikit-learn` `model` field either), then "
-            "no ML updating will be done. Also, if an ML model is provided, no "
-            "nudging will be done. Can be provided multiple times, "
-            "ex. --model_url model1 --model_url model2. If multiple urls are given, "
-            "they will be combined into a single model at runtime, providing the "
-            "outputs are nonoverlapping."
-        ),
-    )
-    parser.add_argument(
```
Comment (AnnaKwa) [inline workflows/prognostic_c48_run/runtime/segmented_run/prepare_config.py:215]: If this info is not in the docs, can you move this info there so it is still somewhere discoverable?

File: workflows/prognostic_c48_run/docs/config-usage.rst [hunk 2]
```diff
@@ -89,7 +87,7 @@ reference analysis data. Here is an example ``minimal.yaml``:
 Machine learning
 ~~~~~~~~~~~~~~~~
 
-A machine learning run can be configured in two ways. The first is by
+A machine learning run is configured by
```
Comment (AnnaKwa) [inline workflows/prognostic_c48_run/docs/config-usage.rst:90]: This is an old error, but can you update this to "one or more models"?

**Review by AnnaKwa — APPROVED** [review]
Thanks, LGTM- can you move the info about the `diagnostic_ml` option into the docs?

File: workflows/prognostic_c48_run/runtime/segmented_run/prepare_config.py (hunk 1 above)
Comment (oliverwm1) [reply workflows/prognostic_c48_run/runtime/segmented_run/prepare_config.py:215]: The option is documented in the `MachineLearningConfig` dataclass, which is included in the API page on the docs.

**Pull request merged**